### PR TITLE
Yaml files for ARE 2015-16 and 2016-17 added.

### PR DESCRIPTION
**Pull request by BarisSari**

### CODE REVIEW

File: 2016-17/provincial/are/pipeline-spec.yaml [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      - name: FY_Descript
+        patterns:
+        - find: ^(Actual Payments)
+          replace: Total
+        - find: ^(Adjusted Appropriation)
+          replace: Total
+        - find: ^(Final Appropriation)
+          replace: Total
+      - name: Department
+        patterns:
```
Comment (jbothma) [inline 2016-17/provincial/are/pipeline-spec.yaml]: mmmm not sure why this was in previous examples - please remove from this one

File: 2015-16/provincial/are/pipeline-spec.yaml [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            replace: "Final Appropriation"
+        - name: FY_Descript
+          patterns:
+          - find: ^(Actual Payments)
+            replace: Total
+          - find: ^(Adjusted Appropriation)
+            replace: Total
+          - find: ^(Final Appropriation)
+            replace: Total
+        - name: Department
```
Comment (jbothma) [inline 2015-16/provincial/are/pipeline-spec.yaml]: mmmm not sure why this was in previous examples - please remove from this one

We shouldn't be rewriting department names directly here, only using the department_names processor

File: 2016-17/provincial/are/pipeline-spec.yaml (hunk 1 above)
Comment (jbothma) [reply 2016-17/provincial/are/pipeline-spec.yaml]: See above

Comment (BarisSari) [reply 2016-17/provincial/are/pipeline-spec.yaml]: Do you want me to remove all of the replacements for FY_Descript or only for Final Appropriation?

File: 2015-16/provincial/are/pipeline-spec.yaml (hunk 2 above)
Comment (BarisSari) [reply 2015-16/provincial/are/pipeline-spec.yaml]: So, do you think that programme_names processor is enough?

File: 2016-17/provincial/are/pipeline-spec.yaml (hunk 1 above)
Comment (jbothma) [reply 2016-17/provincial/are/pipeline-spec.yaml]: No, please keep the replacements of FY_Descript. Only remove replacements of Department

File: 2015-16/provincial/are/pipeline-spec.yaml (hunk 2 above)
Comment (jbothma) [reply 2015-16/provincial/are/pipeline-spec.yaml]: See above - we shouldn't be doing replacements on the Department column in the pipeline. Just remove line 64-71

File: 2016-17/provincial/are/pipeline-spec.yaml (hunk 1 above)
Comment (BarisSari) [reply 2016-17/provincial/are/pipeline-spec.yaml]: But when I do that, I face KeyError exception. Here's an example:

> INFO    :RESULTS:
> INFO    :FAILURE: ./2016-17/provincial/are/are-2016-17 
> ERROR log from processor programme_names:
> +--------
> | ERROR   :Traceback (most recent call last):
> | ERROR   :File "/home/baris/PycharmProjects/treasury-pipelines/programme_names.py", line 87, in <module>
> | ERROR   :process_row=process_row)
> | ERROR   :File "/home/baris/miniconda3/lib/python3.7/site-packages/datapackage_pipelines/wrapper/wrapper.py", line 232, in process
> | ERROR   :process_row)
> | ERROR   :File "/home/baris/miniconda3/lib/python3.7/site-packages/datapackage_pipelines/wrapper/wrapper.py", line 181, in __exit__
> | ERROR   :spew(self.datapackage, self.resource_iterator, stats=self.stats)
> | ERROR   :File "/home/baris/miniconda3/lib/python3.7/site-packages/datapackage_pipelines/wrapper/wrapper.py", line 75, in spew
> | ERROR   :for rec in res:
> | ERROR   :File "/home/baris/miniconda3/lib/python3.7/site-packages/datapackage_pipelines/wrapper/wrapper.py", line 201, in generic_process_resource
> | ERROR   :parameters, stats)
> | ERROR   :File "/home/baris/PycharmProjects/treasury-pipelines/programme_names.py", line 72, in process_row
> | ERROR   := programme_names[sphere][government_name][department_name].\
> | ERROR   :KeyError
> | ERROR   ::
> | ERROR   :'SOCIAL DEVELOPMENT AND SPECIAL PROGRAMMES'
> +--------

Comment (jbothma) [reply 2016-17/provincial/are/pipeline-spec.yaml]: Ah I see - please remove the programme_names processor then.

The reasoning here is, the data we get is dirty, but we don't have the authority to change department names. If it's just a matter of capitalisation we can and do "fix" it with "slugify". But even a spelling mistake we're not fixing at the moment. Instead, when there are discrepancies between department or programme names when comparisons are done later, our government can see the issue and fix their source data.

Thanks for trying though.

Comment (BarisSari) [reply 2016-17/provincial/are/pipeline-spec.yaml]: Okay, I understand. You can find the changed versions of two files. When whole process is done, I'm going to send you the results as well.

https://github.com/vulekamali/treasury-pipelines/pull/25/commits/9eca39925701ff85e34c1e81d98675d8914119a4

File: 2015-16/provincial/are/pipeline-spec.yaml (hunk 2 above)
Comment (BarisSari) [reply 2015-16/provincial/are/pipeline-spec.yaml]: Okay, I understand. You can find the changed versions of two files. When whole process is done, I'm going to send you the results as well.

https://github.com/vulekamali/treasury-pipelines/pull/25/commits/9eca39925701ff85e34c1e81d98675d8914119a4

**Pull request merged**